### PR TITLE
Deprecate Multiplayer "Session" Events

### PR DESCRIPTION
**Pull request by AMZN-Gene**
- Deprecate IMultiplayer SessionInit and SessionShutdown events. It's not being used, and confusing as there's already SessionNotificationBus which developers should be using instead
- Rename sv_useGameLiftServer to sv_gameLiftEnabled to follow cvar naming convention 

Deprecation GHI: https://github.com/o3de/o3de/issues/15690
Original "Bug": #15626

### CODE REVIEW

**Review by kberg-amzn — APPROVED** [review]
Looks good. I would remove the deprecated methods from dev immediately..

File: Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp [hunk 1]
```diff
@@ -186,13 +186,13 @@ namespace Multiplayer
                     mpComponent->GetNetworkEntityManager()->ClearAllEntities();
                 })
                 ->Method("GetCurrentBlendFactor", []()
+                {
+                    if (GetMultiplayer())
```
Comment (AMZN-Olex) [inline Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp:190]: You might as well save a pointer here, each GetMultiplayer goes through a mutex due to using AZ::Interface.

**Review by AMZN-Olex — APPROVED** [review]

File: Gems/AWSGameLift/Code/AWSGameLiftServer/Source/AWSGameLiftServerSystemComponent.cpp [hunk 2]
```diff
@@ -19,7 +19,7 @@
 
 namespace AWSGameLift
 {
-    AZ_CVAR(bool, sv_useGameLiftServer, false, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Activate GameLift server manager and SDK");
+    AZ_CVAR(bool, sv_gameLiftEnabled, false, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Activate GameLift server manager and SDK");
```
Comment (allisaurus) [inline Gems/AWSGameLift/Code/AWSGameLiftServer/Source/AWSGameLiftServerSystemComponent.cpp:22]: is there a corresponding o3de.org docs update to reflect this change?

File: Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp (hunk 1 above)
Comment (AMZN-Gene) [reply Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp]: Thank you! Fixed.

File: Gems/AWSGameLift/Code/AWSGameLiftServer/Source/AWSGameLiftServerSystemComponent.cpp (hunk 2 above)
Comment (AMZN-Gene) [reply Gems/AWSGameLift/Code/AWSGameLiftServer/Source/AWSGameLiftServerSystemComponent.cpp:22]: Now there is... thanks!
https://github.com/o3de/o3de.org/pull/2357

**Review by allisaurus — APPROVED** [review]

**Pull request merged**